core/src/apps/management/recovery_device/__init__.py

```diff
@@ -31,68 +33,75 @@ async def recovery_device(msg: RecoveryDevice) -> Success:
 
     from .homescreen import recovery_homescreen, recovery_process
 
-    dry_run = msg.dry_run  # local_cache_attribute
+    recovery_kind = msg.kind or RecoveryKind.NormalRecovery  # local_cache_attribute
 
     # --------------------------------------------------------
     # validate
-    if not dry_run and storage_device.is_initialized():
-        raise wire.UnexpectedMessage("Already initialized")
-    if dry_run and not storage_device.is_initialized():
-        raise wire.NotInitialized("Device is not initialized")
+    if recovery_kind == RecoveryKind.NormalRecovery:
+        if storage_device.is_initialized():
+            raise wire.UnexpectedMessage("Already initialized")
+    elif recovery_kind in (RecoveryKind.DryRun, RecoveryKind.UnlockRepeatedBackup):
+        if not storage_device.is_initialized():
+            raise wire.NotInitialized("Device is not initialized")
+        # check that only allowed fields are set
+        for key, value in msg.__dict__.items():
+            if key not in DRY_RUN_ALLOWED_FIELDS and value is not None:
+                raise wire.ProcessError(
+                    f"Forbidden field set in dry-run: {key}"
+                )
+    else:
+        raise ValueError("Unknown RecoveryKind")
```

Comment:
```suggestion
        raise RuntimeError  # Unknown RecoveryKind
```

this will not actually happen because the protobuf layer will refuse to decode to invalid value.
per style guide, we don't add explanatory arguments to builtin exceptions.

Reply:
https://github.com/trezor/trezor-firmware/pull/3743/commits/65fef0e3af060fb95245121b8d6e50a194dad8cb#diff-1751448e00180998cfe48dbfeb7a35ffb20d53c8938f071a5b44eb456a5f8943R57